core/src/main/java/io/cucumber/core/plugin/JSONFormatter.java

```diff
@@ -212,6 +222,11 @@ private void finishReport() {
         return testCaseMap;
     }
 
+    private String getDateTimeFromTimeStamp(TestCaseStarted event) {
+        LocalDateTime localDateTime = LocalDateTime.ofInstant(Instant.ofEpochMilli(event.getElapsedTimiMillis()), ZoneId.systemDefault());
+        return localDateTime.format(dateTimeFormatter == null ? DateTimeFormatter.ISO_DATE_TIME : dateTimeFormatter);
```

Comment:
Is there a reason the `dateTimeFormatter` is optional?

Reply:
For the JSON output the format should be fixed predictable. So I would think the zone should always be UTC. Using a `ZonedDateTime` may be more appropriate but I'm not sure.

Reply:
And probably good to bear in mind that v4 is still using Java 7. So you can't use the fancy Date stuff yet.

Reply:
And well Java 7 puts away both the localdatetime and zonedatetime options away.
I ll change it to the usual Date.

And shouldn't it just be the default zone of the PC running the TCs? In case they use it to see the timestamp of the TC run in the report.

Reply:
I kept the dateTimeFormatter optional for the sake of flexibility in case we dont want the full time for some TC (we just want the date, etc).

Its actually also being used in the JUnits to bypass the 'dynamic generation' issue, for eg in `RuntimeTest`, `JsonParallelRuntimeTest`. I can make it constant but I ll need some suggestion on how to update the JUnits then. Do u have an idea?

Reply:
I'd prefer not to depend on the default timezone of the machine running the tests, as this might be either a local machine with locally running application, or a CI server (i.e. Jenkins) running tests against an actual environment; where the CI server and the test environment might not have the same system datetime / timezone. Being able to specify it is preferable inho

Reply:
Yes that makes sense. I can change it to always be UTC in that case.

But do u mean allow the user to specify what timezone to use before running the tests? From cucumber options or the like?